planner/core/prepare_test.go

```diff
@@ -42,6 +42,80 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestPreparePointGetWithDML(t *testing.T) {
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+	orgEnable := core.PreparedPlanCacheEnabled()
+	defer core.SetPreparedPlanCache(orgEnable)
+	core.SetPreparedPlanCache(true)
+	se, err := session.CreateSession4TestWithOpt(store, &session.Opt{
+		PreparedPlanCache: kvcache.NewSimpleLRUCache(100, 0.1, math.MaxUint64),
+	})
+	require.NoError(t, err)
+	tk := testkit.NewTestKitWithSession(t, store, se)
+
+	tk.MustExec(`use test`)
+	tk.MustExec(`drop table if exists t`)
+	tk.MustExec(`create table t(a int, unique key(a))`)
+	tk.MustExec(`insert into t values(1), (2)`)
+
+	// txn1 left a cached plan
+	tk.MustExec(`begin`)
+	tk.MustExec(`prepare stmt from 'update t set a = ? where  a = ?'`)
+	tk.MustExec(`set @a=1`)
+	tk.MustExec(`execute stmt using @a, @a`)
+	tk.MustExec(`commit`)
+
+	// txn2 can reuse the cached plan generated by txn1 directly
+	tk.MustExec(`begin`)
+	tk.MustExec(`prepare stmt from 'update t set a = ? where  a = ?'`)
+	tk.MustExec(`set @a=2`)
+	tk.MustExec(`execute stmt using @a, @a`) // can reuse the cached plan directly
+	tk.MustQuery(`select @@last_plan_from_cache`).Check(testkit.Rows("1"))
+	tk.MustExec(`rollback`)
+}
+
+func TestPrepareIgnoreCloseStmtCmd(t *testing.T) {
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+	orgEnable := core.PreparedPlanCacheEnabled()
+	defer core.SetPreparedPlanCache(orgEnable)
+	core.SetPreparedPlanCache(true)
+	se, err := session.CreateSession4TestWithOpt(store, &session.Opt{
+		PreparedPlanCache: kvcache.NewSimpleLRUCache(100, 0.1, math.MaxUint64),
+	})
+	require.NoError(t, err)
+	tk := testkit.NewTestKitWithSession(t, store, se)
+
+	tk.MustExec(`use test`)
+	tk.MustExec(`drop table if exists t`)
+	tk.MustExec(`create table t(a int, unique key(a))`)
+
+	// disable the ignore-stmt-cmd
```

Comment:
I check the binary protocol locally and it can work:
![image](https://user-images.githubusercontent.com/7499936/157251718-b204e1b7-535f-46e8-9c5a-c8ff920a3bf3.png)